jina/executors/indexers/helper.py

```diff
@@ -0,0 +1,52 @@
+from typing import Iterator
+from . import BaseIndexer
+
+
+class DuplicateChecker(BaseIndexer):
+
+    class EmptyHandler:
+        def __init__(self):
+            pass
+
+        def close(self):
+            pass
+
+        def flush(self):
+            pass
+
+        def write(self, *args, **kwargs):
+            pass
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(index_filename='', *args, **kwargs)
+        self.indexed = set()
+        self._size = self._indexed_size
+
+    def _filter_ids(self, id_list):
+        is_indexed = []
+        for id in id_list:
+            _indexed = (id in self.indexed)
+            is_indexed.append(_indexed)
+            if not _indexed:
+                self.indexed.add(id)
+        return is_indexed
+
+    def add(self, ids: Iterator[int], *args, **kwargs):
+        is_indexed = self._filter_ids(ids)
+        return is_indexed
```

Comment:
I'm not a fan of this part. Can we use on-disk storage. Please note that `KVIndexer` and `BinaryPbIndexer` spend quite some effort on *not storing any data in memory*, hence their memory consumption is completely constant during indexing.

Reply:
Definitely, I've listed this in the **Remaining Issues** and will start working on this.